.travis.yml

```diff
@@ -18,8 +18,21 @@ matrix:
       env: RUN_COVERAGE=false TZ=UTC
     - python: 3.8-dev
       env: RUN_COVERAGE=false
+    - os: osx
+      env: RUN_COVERAGE=false
+      # Install Python via Homebrew, Travis cannot install Python ATM.
+      language: generic
```

Comment:
> The MacOS stage says "no language set" instead of "python ..."?

That would be because of this line.

**tldr;** *Travis can't install different python versions automatically you need to provision it yourself.*

**long overdue rant;** As is well known Python environments usually become a bit of a mess. On Mac OS the problem becomes slightly messier as Mac OS does not actually have a package manager, well not an official one anyway. Apple gives you system Python but if you want anything else there isn't an Apple provided solution to getting it. The on-official but incredibly popular Brew package manager might as well be dubbed "the" Mac OS package manager. Most people turn to Brew to install different Python versions, consequently the Brew formula is pretty good and actually surpasses the system installation in some ways. So Brew Python is what everyone should use but neither it nor Brew are actually installed by default leaving Travis in a slightly sticky position. Their solution for the moment is to do nothing which is great except that if you try to specify a Python version it just dies in a nonsensical way. The workaround to all this mess is to set the language to something other than Python and just install Brew Python yourself.